Mage.Sets/src/mage/cards/a/ApproachOfTheSecondSun.java

```diff
@@ -64,28 +64,30 @@ public ApproachOfTheSecondSunEffect copy() {
     @Override
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
-        Spell spell = game.getStack().getSpell(source.getSourceId(), false);
-        if (controller != null && spell != null) {
-            ApproachOfTheSecondSunWatcher watcher
-                    = game.getState().getWatcher(ApproachOfTheSecondSunWatcher.class);
-            if (watcher != null
-                    && watcher.getApproachesCast(controller.getId()) > 1
-                    && spell.getFromZone() == Zone.HAND) {
-                // Win the game
-                controller.won(game);
-            } else {
-                // Gain 7 life and put this back into library.
-                controller.gainLife(7, game, source);
-
-                // Put this into the library as the 7th from the top
-                Card spellCard = game.getStack().getSpell(source.getSourceId(), false).getCard();
-                if (spellCard != null) {
-                    controller.putCardOnTopXOfLibrary(spellCard, game, source, 7, true);
-                }
+        Spell spell = game.getStack().getSpell(source.getSourceId());
+        ApproachOfTheSecondSunWatcher watcher
+                = game.getState().getWatcher(ApproachOfTheSecondSunWatcher.class);
+        if (controller == null || spell == null || watcher == null) {
+            return false;
+        }
+        //If this spell was cast from your hand and you've cast another spell named {this} this game
```

Comment:
Nitpicky, but can you please make sure to leave the space between the "//" and the comment.